static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -57,36 +127,115 @@ function SelectProjectStep({
   Body: ModalBody,
   Header: ModalHeader,
   Footer: ModalFooter,
+  closeModal,
   toStep,
   step,
+  project,
+  setProject,
 }: OnboardingStepProps) {
-  const onNext = useCallback(
-    (platform: 'iOS' | 'Android') => {
+  const {projects} = useProjects();
+
+  const onFormSubmit = useCallback(
+    (evt: React.FormEvent) => {
+      evt.preventDefault();
+
+      if (!project?.platform) {
+        return;
+      }
+
+      const nextStep = platformToInstructionsMapping[project.platform];
+      if (nextStep === undefined) {
+        // We should probably be throwing here
```

Comment:
can we throw here? silently returning might make bugs hard to track down in the future

Reply:
Yes, we just need to be careful because setting disabled on the select option doesnt actually prevent the user from selecting an unsupported option - hence why I didnt want random errors to be throw around if for some weird reason the input triggers for submit :/

Reply:
oh really 🤔 i think we need to address that at some point